### ISSUE TITLE
Unable to select a block style variation in the side panel when isDefault property is used

### ISSUE THREAD

**Issue by skorasaurus** [WordPress/gutenberg#23211]
**Describe the bug**

Beginning in Gutenberg 8.2.0, 
I'm unable to select a custom block style variation that I made (to the button block) in the side panel when I try to click on it or use a keypress to activate it. 

here's a short video of it happening: 
https://f002.backblazeb2.com/file/cpl-public/tech-devel/2020-06-14-unable-to-select-block-style-variation.ogv 

This behavior happens across different themes; including twentytwenty.  

(My block style variation is stored in a plugin specifically (https://gitlab.com/cpl/cpl-blocks/-/blob/eb415df5076334090bfdd6046b9685005a33e46f/src/block/button.js) and hasn't been modified in a month); 

You can download and run the plugin at https://gitlab.com/cpl/cpl-blocks/-/tree/eb415df5076334090bfdd6046b9685005a33e46f
(it is based off create-guten-block)

A smaller test case is available at https://gitlab.com/cpl/cpl-test-case-button-take-two/ 
( just enable the plugin)

This regression was introduced some time between Gutenberg 8.1.0 (where the problem does not exist) and Gutenberg 8.2. I rebuilt Gutenberg 8.1.0 in a test environment (from commit fc6ed3e17b441d7584a32985ebdbcb1f47199706) and was unable to replicate the problem. 

This problem does not occur when the gutenberg plugin is disabled and the Gutenberg included in  
(WordPress 5.4.2) is used.  I've been able to replicate this problem using Gutenberg that was built from the master branch as recently as June 13th. 

**To reproduce**
1. download https://gitlab.com/cpl/cpl-test-case-button-take-two/ 
2. activate 
3. create new page 
4, add button block
5. attempt to add the block style variation by clicking on the "cpl default link action" in the side panel
(notice that no errors appears in the console). 

**Expected behavior**
Be able to select and apply the custom style variation 

**Screenshots**
https://f002.backblazeb2.com/file/cpl-public/tech-devel/2020-06-14-unable-to-select-block-style-variation.ogv 

**Editor version (please complete the following information):**
- WordPress version: 5.4.x
- Does the website has Gutenberg plugin installed: yes, versions 5.2.0+

**Comment by oxyc** [comment]
I can reproduce by registering it in the developer console:

```javascript
wp.blocks.registerBlockStyle( 'core/button', {
    name: 'cpl-button--action',
    label: 'CPL\'s Link default action ',
    isDefault: true
} );
```

If I remove `isDefault` it works fine.

**Comment by skorasaurus** [comment]
Thank you @oxyc ! I thought I had tried removing `isDefault` and it didn't work but I must have done something incorrect. The block style variation is able to be loaded as intended when I remove the `isDefault` property :) 

As I understand, having the `isDefault` property set to true means that when the block is first loaded in the editor, that style (or style variation) will already be selected in the sidepanel. 

With this in mind, I would like the class name to always be present; regardless if it has `isDefault: true`. 

In our case, we're (trying) to use BEM with Gutenberg and the class that I'd like to apply as default is a [modifier](https://en.bem.info/methodology/quick-start/#modifier) that we use very often (for the sake of assisting our content editors, that modifer is already pre-selected for them). 

However, [decisions have been to not add the class by default to the html output](https://github.com/WordPress/gutenberg/commit/a00c7f4b028c556e38d174706b4e2b3fb35e96bc) (also see https://github.com/WordPress/gutenberg/pull/22266)

**Comment by talldan** [comment]
Removing the regression label since it looks like this wasn't a regression.

**Comment by oxyc** [comment]
> It feels like a new feature and needs more thought.

If not made into a feature, users who need it can still do it fairly easily using the `editor.BlockListBlock` filter.

**Comment by talldan** [comment]
I did some more testing on 5.4.2, and I've changed my opinion and now consider this a regression. What confused me originally is that the button that's added initially by the buttons block is unaffected by registering a new default because it's created using a template that has no class names specified:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/buttons/edit.js#L16

But subsequently added button blocks do have both the style and classes applied from the new default.

Also, thinking more about what I've proposed in #23512 I see some problems. One is that the default block styles core supplies rely on the absence of a classname, while the block styles themes supply typically don't. I'm not sure how themes could affect the styling of a block if the classname from a block style isn't applied.

Given the short timescales for 5.5, I think the safest option would be to revert #22266, and then work on improving the feature for 5.6.
